### PR TITLE
Verify unique id signature at WPE

### PR DESCRIPTION
**Pull request by manju956**
- Verify verification key signature which is digital signature taken on
  hash of concatenation of unique id and nonce(WPE generated).

Signed-off-by: manju956 <manjunath.a.c@intel.com>

### CODE REVIEW

File: tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp [hunk 1]
```diff
@@ -75,6 +75,41 @@ tcf_err_t ecall_GenerateNonce(uint8_t* out_nonce, size_t in_nonce_size) {
     return result;
 }  // ecall_GenerateNonce
 
+tcf_err_t ecall_VerifyUniqueIdSignature(const char* unique_id,
```
Comment (manojgop) [inline tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp:78]: unique_id -> unique_id_key. Just to emphasize that the unique id is a key. You could add a description what this function is doing and the parameters

File: enclave_manager/avalon_enclave_manager/wpe/wpe_enclave_manager.py [hunk 2]
```diff
@@ -64,9 +64,20 @@ def _create_signup_data(self):
...
+            return None
         # Received response contains result,verification_key and
         # verification_key_signature delimited by ' '
         self._unique_verification_key = response.split(' ')[1]
+        self._unique_verification_key_signature = response.split(' ')[2]
+        # Verify unique verification key signature using unique id
+        result = signup_cpp_obj.VerifyUniqueIdSignature(
+            self._unique_verification_key,
+            self._unique_verification_key_signature)
+        if result != 0:
```
Comment (Ram-srini) [inline enclave_manager/avalon_enclave_manager/wpe/wpe_enclave_manager.py:78]: Compare against constant TCF_SUCCESS.

File: tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp (hunk 1 above)
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp]: Done

File: enclave_manager/avalon_enclave_manager/wpe/wpe_enclave_manager.py (hunk 2 above)
Comment (manju956) [reply enclave_manager/avalon_enclave_manager/wpe/wpe_enclave_manager.py:78]: To compare with enclave error enums, we have to expose enclave related enum types via swig interface. This way of exposing was not encouraged in some previous PR.

File: tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp [hunk 3]
```diff
@@ -75,6 +78,57 @@ tcf_err_t ecall_GenerateNonce(uint8_t* out_nonce, size_t in_nonce_size) {
...
+        std::string b64_verify_key = ByteArrayToStr(verify_key_hex_bytes);
+        tcf::crypto::sig::PublicKey public_key(b64_verify_key);
+
+        EnclaveData* enclave_data = EnclaveData::getInstance();
+        // compute hash on b64 encoded verification key and nonce hex string
+        std::string concat_str = b64_verify_key + enclave_data->get_nonce();
+        ByteArray hash = tcf::crypto::ComputeMessageHash(
+            StrToByteArray(concat_str));
+        std::string b64_verify_key_sig = ByteArrayToStr(
+            HexEncodedStringToByteArray(verification_key_signature));
```
Comment (manojgop) [inline tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp:110]: `b64_verify_key_sig`  is unused. Please remove

File: tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp [hunk 4]
```diff
@@ -75,6 +78,57 @@ tcf_err_t ecall_GenerateNonce(uint8_t* out_nonce, size_t in_nonce_size) {
...
+        }
+    } catch (tcf::error::Error& e) {
+        SAFE_LOG(TCF_LOG_ERROR,
+            "Error in ecall_VerifyUniqueIdSignature: %04X -- %s",
+            e.error_code(), e.what());
+        ocall_SetErrorMessage(e.what());
+        result = e.error_code();
+    } catch (...) {
+        SAFE_LOG(TCF_LOG_ERROR,
+            "Unknown error in Avalon enclave(ecall_CreateSignupDataWPE)");
```
Comment (manojgop) [inline tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp:126]: ecall_CreateSignupDataWPE -> ecall_VerifyUniqueIdSignature

File: tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/wpe/signup_wpe.cpp [hunk 5]
```diff
@@ -73,6 +73,58 @@ tcf_err_t SignupDataWPE::GenerateNonce(
     return result;
 }  // SignupDataWPE::GenerateNonce
 
+/* Verify unique id singature obtained from KME by using unique id key.
```
Comment (manojgop) [inline tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/wpe/signup_wpe.cpp:76]: singature  -> signature

File: tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp (hunk 3 above)
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp]: Removed.

File: tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp (hunk 4 above)
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave/wpe/signup_enclave_wpe.cpp]: Done

File: tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/wpe/signup_wpe.cpp (hunk 5 above)
Comment (manju956) [reply tc/sgx/trusted_worker_manager/enclave_untrusted/enclave_bridge/wpe/signup_wpe.cpp]: Done

**Pull request merged**